src/crypto/CHIPCryptoPAL.h

```diff
@@ -214,6 +262,426 @@ CHIP_ERROR add_entropy_source(entropy_source fn_source, void * p_source, size_t
+     **/
+    virtual CHIP_ERROR KDF(const unsigned char * ikm, const size_t ikm_len, const unsigned char * salt, const size_t salt_len,
+                           const unsigned char * info, const size_t info_len, unsigned char * out, size_t out_len) = 0;
+    /**
+     * @brief Load a field element.
+     *
+     *  @param in     The input big endian field element.
+     *  @param in_len The size of the input buffer in bytes.
+     *  @param fe     A pointer to an initialized implementation dependant field element.
+     *
+     *  @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR FELoad(const unsigned char * in, size_t in_len, void * fe) = 0;
+
+    /**
+     * @brief Write a field element in big-endian format.
+     *
+     *  @param fe      The field element to write.
+     *  @param out     The output buffer.
+     *  @param out_len The length of the output buffer.
+     *
+     *  @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR FEWrite(const void * fe, unsigned char * out, size_t out_len) = 0;
+
+    /**
+     * @brief Generate a field element.
+     *
+     *  @param fe  A pointer to an initialized implementation dependant field element.
+     *
+     *  @requirements The implementation must generate a random element from [0, q) where q is the curve order.
+     *
+     *  @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR FEGenerate(void * fe) = 0;
+
+    /**
+     * @brief Multiply two field elements, fer = fe1 * fe2.
+     *
+     *  @param fer   A pointer to an initialized implementation dependant field element.
+     *  @param fe1  A pointer to an initialized implementation dependant field element.
+     *  @param fe2  A pointer to an initialized implementation dependant field element.
+     *
+     *  @requirements The result must be a field element (i.e. reduced by the curve order).
+     *
+     *  @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR FEMul(void * fer, const void * fe1, const void * fe2) = 0;
+
+    /**
+     * @brief Load a point from 0x04 || X || Y format
+     *
+     * @param in     Input buffer
+     * @param in_len Input buffer length
+     * @param R      A pointer to an initialized implementation dependant point.
+     *
+     *  @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR PointLoad(const unsigned char * in, size_t in_len, void * R) = 0;
+
+    /**
+     * @brief Write a point in 0x04 || X || Y format
+     *
+     * @param R       A pointer to an initialized implementation dependant point.
+     * @param out     Output buffer
+     * @param out_len Length of the output buffer
+     *
+     *  @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR PointWrite(const void * R, unsigned char * out, size_t out_len) = 0;
```

Comment:
nit: maybe having a typedef for the void* would increase readability. Thinking of:

PointWrite(const Point R, unsinged char * out, size_t out_len) 

as being faster to digest.